algo-builder-tester/assets/price-verifier.py

```diff
@@ -0,0 +1,78 @@
+# code like TST1 is to relate the test file `algob-tester.ts`.
+# Add directory to path so that algobpy can be imported
+from math import ceil
+import sys
+
+sys.path.insert(0, ".")
+
+from algobpy.parse import parse_params
+from pyteal import *  # type: ignore wildcard import
+
+STATIC_PRICE = 3456789
+last_price = Bytes("last_price")
+
+
+def price_verifier():
+    """
+    This supports algob-tester, to get external global state
+    """
+    SuccessSeq = Seq(
+        # Log(Bytes("success")),
+        Return(Int(1)),
+    )
+    on_creation = Seq(
+        App.globalPut(last_price, Int(STATIC_PRICE)),
+        App.globalPut(Bytes("last_UTC0"), Bytes("1234567890-1234")),
+        SuccessSeq,
+    )
+    on_opt_in = SuccessSeq
+    on_close_out = SuccessSeq
+    on_update_app = SuccessSeq
+    on_delete_app = SuccessSeq
+    uint = ScratchVar(TealType.bytes)
+    on_call = Seq(
+        uint.store(Txn.application_args[0]),
+        # TODO:discuss#1: how to pass Int?
+        # appArgs: ["str:10348967n", "int:10348967"],
+        # appArgs: [new Uint8Array(10348967), "int:10348967"],
+        # If(
+        #     Txn.application_args[0] == Txn.application_args[1],
+        #     Log(Bytes("same")),
+        #     Log(Bytes("not same")),
+        # ),
+        App.globalPut(last_price, Int(10348967)),
+        App.globalPut(Bytes("last_UTC0"), Gtxn[0].application_args[1]),
```

Comment:
Is there one price-verifier contract per asset?

Reply:
I didn't think about this. Since the functionality is uniform among all assets. I think we should write them as one contract, right?

Reply:
If it's one contract for all, you mean deploy only one price-verifier contract for whole system. What if two user use the system at the same time? Also, I think we need to store the ASA address to tell which is this price for. To address the multiple user use at same time problem, the ASA address, price and timestamp should be stored as local state

Reply:
^ seems a major functionality issue, so I'll mark the PR as "changes requested"

Reply:
These problems you said are definitely needed to be taken care of. But this branch is only a functionality test and not the real `price-verifier` for us to use. I would prefer to write a `price-verifier.py` contract in the python package in a new feature branch. We should do some design docs later for the `price-verifier` for milestone 2 later. What do you think about my idea?